code/modules/mob/living/carbon/alien/humanoid/queen.dm

```diff
@@ -68,6 +68,14 @@
 	internal_organs += new /obj/item/organ/alien/eggsac
 	..()
 
+/mob/living/carbon/alien/humanoid/royal/queen/proc/check_hostile()
+	if(is_station_level(src.z)) //we don't want the hostile environment if the xenos aren't actually on station
+		SSshuttle.registerHostileEnvironment(src) //aliens delay shuttle
+		addtimer(CALLBACK(src, .proc/game_end), 30 MINUTES) //time until shuttle is freed/called
+		return
```

Comment:
you should check if hostile environment isn't already registered

Reply:
Talked about this on discord, can I consider this resolved? 

This should never come up.